src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -0,0 +1,974 @@
+            {{customNodeOutputName, pipelineOutputName + std::to_string(i)}});
+        pipeline.push(std::move(custom_nodes[i]));
+    }
+    pipeline.push(std::move(input_node));
+    pipeline.push(std::move(output_node));
+
+    ASSERT_EQ(pipeline.execute(), StatusCode::OK);
+    ASSERT_EQ(response.outputs().size(), N);
+
+    for (int i = 0; i < N; i++) {
+        this->checkResponse<float>(
+            pipelineOutputName + std::to_string(i),
+            inputValues,
+            [i, addValue, subValue](float value) -> float {
+                value += addValue[i % PARAMETERS];
+                value -= subValue[i % PARAMETERS];
+                return value;
+            });
+    }
+}
+
+TEST_F(EnsembleFlowCustomNodePipelineExecutionTest, CustomAndDLNodes) {
+    // input  add-sub1 dummy  add-sub2 output
+    //  O------->O------O--------O------>O
+    ConstructorEnabledModelManager modelManager;
+    ModelConfig config = DUMMY_MODEL_CONFIG;
+    modelManager.reloadModelWithVersions(config);
+
+    const std::vector<float> inputValues{
+        4, 1.5, -5, -2.5, 9.3, 0.3, -0.15, 7.4, 5.2, -2.4};
+    this->prepareRequest(inputValues);
+
+    const float addValue[] = {-0.85, 30.2};
+    const float subValue[] = {1.35, -28.5};
+
+    auto input_node = std::make_unique<EntryNode>(&request);
+    auto output_node = std::make_unique<ExitNode>(&response);
+    auto model_node = std::make_unique<DLNode>(
+        "dummy_node",
+        "dummy",
+        std::nullopt,
+        modelManager);
+    std::unique_ptr<CustomNode> custom_node[] = {
+        std::make_unique<CustomNode>(customNodeName + "_0", library,
+            parameters_t{
+                {"add_value", std::to_string(addValue[0])},
+                {"sub_value", std::to_string(subValue[0])}}),
+        std::make_unique<CustomNode>(customNodeName + "_1", library,
+            parameters_t{
+                {"add_value", std::to_string(addValue[1])},
+                {"sub_value", std::to_string(subValue[1])}})};
+
+    Pipeline pipeline(*input_node, *output_node);
+    pipeline.connect(*input_node, *(custom_node[0]), {{pipelineInputName, customNodeInputName}});
+    pipeline.connect(*(custom_node[0]), *model_node, {{customNodeOutputName, DUMMY_MODEL_INPUT_NAME}});
+    pipeline.connect(*model_node, *(custom_node[1]), {{DUMMY_MODEL_OUTPUT_NAME, customNodeInputName}});
+    pipeline.connect(*(custom_node[1]), *output_node, {{customNodeOutputName, pipelineOutputName}});
+
+    pipeline.push(std::move(input_node));
+    pipeline.push(std::move(custom_node[0]));
+    pipeline.push(std::move(custom_node[1]));
+    pipeline.push(std::move(model_node));
+    pipeline.push(std::move(output_node));
+
+    ASSERT_EQ(pipeline.execute(), StatusCode::OK);
+    ASSERT_EQ(response.outputs().size(), 1);
+
+    this->checkResponse<float>(inputValues, [addValue, subValue](float value) -> float {
+        return value + 1.0 + addValue[0] + addValue[1] - subValue[0] - subValue[1];
```

Comment:
This 1.0 at first glance looks like magic constant. Maybe define somewhere above int numberOfDummyNodes = 1 and then multiply this here *1.0?

Reply:
I think it would be good to define 1.0 as DUMMY_ADDITION_VALUE=1.0 in test_utils.hpp.